src/domain/impl/RoundRobin.java

```diff
@@ -0,0 +1,56 @@
+package domain.impl;
+
+import java.util.ArrayDeque;
+import java.util.NoSuchElementException;
+
+import domain.api.SchedulingStrategy;
+
+public class RoundRobin extends SchedulingStrategy {
+    public RoundRobin() {
+        ready = new ArrayDeque<>();
+        blocked = new ArrayDeque<>();
+    }
+
+    @Override
+    public void execute() {
+        try {
+            Thread.sleep(quantumSizeMs);
+        } catch (Exception e) {
```

Comment:
Catch specific exceptions instead of generic ones, and avoid rethrowing in new Error objects. You can just write `e.printStackTrace()` for example